frontend/src/static/js/main.js

```diff
@@ -169,6 +174,41 @@ window.app = new window.Vue({
         document.getElementById('tabs-wrapper').scrollTop = 0;
       }
     },
+    renderAuthorShipTabHash(minDate, maxDate) {
+      const hash = window.hashParams;
+      const info = {
+        author: hash.tabAuthor,
+        location: hash.tabLocation,
+        branch: hash.tabBranch,
+        minDate,
+        maxDate,
+      };
+      this.updateInfoRepoName(info);
+      const tabInfoLength = Object.values(info).filter((x) => x).length;
+      if (this.isValidInfo(info)) {
+        this.updateTabAuthorship(info);
+      } else if (hash.tabOpen === 'false' || tabInfoLength > 2) {
+        window.app.isTabActive = false;
+      }
+    },
+    isValidInfo(info) {
+      const tabInfoLength = Object.values(info).filter((x) => x).length;
+      return Object.keys(info).length === tabInfoLength && info.location && info.branch && info.location.split('github.com/').length >= 2;
+    },
+    updateInfoRepoName(info) {
+      if (!this.isValidInfo(info)) {
+        info.repo = '';
+        return;
+      }
+      const repoName = info.location.split('github.com/')[1].split('/');
+      if (repoName.length < 2) {
```

Comment:
isnt this redundant since it has been checked in `isValidInfo`?

Reply:
`isValidInfo`: checks that the url contains `github.com/`
`updateInfoRepoName`: checks that the post-processed url has a _organization_ and a _repository name_.

Reply:
I see. By the way I just recalled that this implementation `info.location.split('github.com/')[1].split('/');` won't work with local analysis because the location won't have github url. 

Possibly to replace `hash.tabLocation` with `organization` instead?

Reply:
For local repos, the `repo name` contains `repo`_`branch`, so there is no `organization`.
An alternative is to hash the `repo name`.